internal/store/bigtable/stat_reader.go

```diff
@@ -55,17 +55,27 @@ func ReadStats(
 			result[psv.Place] = map[string]*model.ObsTimeSeries{}
 		}
 		if _, ok := result[psv.StatVar]; !ok {
-			result[psv.Place][psv.StatVar] = nil
+			result[psv.Place][psv.StatVar] = &model.ObsTimeSeries{}
 		}
 	}
 	for _, rowKey := range rowList {
 		token, _ := keyToTokenFn(rowKey)
 		psv := keyTokens[rowKey]
-		if data, ok := branchDataMap[token]; ok {
-			result[psv.Place][psv.StatVar] = data.(*model.ObsTimeSeries)
-		} else if data, ok := baseDataMap[token]; ok {
-			result[psv.Place][psv.StatVar] = data.(*model.ObsTimeSeries)
+		// Different base data has different source series, concatenate them together.
+		ss := result[psv.Place][psv.StatVar].SourceSeries
```

Comment:
So this assumes there must be no overlap?

Reply:
Yep, for the base import groups. I will move the merging logic into separate functions, and after we introduce the branch import group, will need to add the capability of picking from preferred group.